.github/workflows/release.yml

```diff
@@ -0,0 +1,72 @@
+name: Release
+
+on:
+  push:
+    # Sequence of patterns matched against refs/tags
+    tags:
+      - 'v*' # Push events to matching v*, i.e. v1.0, v20.15.10
+
+jobs:
+  release:
+    runs-on: macos-latest
+    
+    steps:
+      - uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # This token is provided by Actions, you do not need to create your own token
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: Release ${{ github.ref }}
+          draft: false
+          prerelease: false
+        id: create_release
+
+      - run: echo "${{ steps.create_release.outputs.upload_url }}" > upload_url.txt
+
+      - uses: actions/upload-artifact@v1
+        with:
+          name: upload_url
+          path: upload_url.txt
+
+  build:
+    runs-on: macos-latest
+
+    needs: [ release ]
+
+    strategy:
+      matrix:
+        go: [ '1.13' ]
+        goos: [ 'linux', 'darwin' ]
```

Comment:
Does this mean that all of the release builds happen on macOS, and the Linux build gets cross-compiled? I would be worried that this won't work with cgo - in particular, the go-keychain package uses cgo, and it's not possible to cross-compile a macOS binary from a Linux builder. It might be possible to produce a Linux binary from macOS though...

If it's difficult to do this, and you can confirm that it produces a working Linux binary, then we can probably leave it and deal with it later (i.e. when we introduce a cgo package that doesn't build on macOS).

Reply:
Thanks for this. I've changed it to build on the native os. There was an issue with crossbuild with:

```
        env:
          GOOS: ${{ matrix.target.goos }}
          GOARCH: ${{ matrix.target.goarch }}
          CGO_ENABLED: 1
```

I've added:

```
    strategy:
      matrix:
        target:
          - os: 'macos-latest'
            goos: 'darwin'
            goarch: 'amd64'
          - os: 'ubuntu-16.04'
            goos: 'linux'
            goarch: 'amd64'
        go: [ '1.13' ]
  
    runs-on: ${{ matrix.target.os }}
```

and it looks good.